src/render.ts

```diff
@@ -63,29 +63,41 @@ export default function render(
         .attr('font-weight', fontWeight)
         .attr('text-anchor', 'middle')
         .attr('transform', 'translate(0, 0) rotate(0)')
-        .call(enter =>
+        .call(enter => (enableTransition) ?
```

Comment:
Let's do it this way if `transitionDuration = 0` creates issues that you mentioned:


```js
.call(enter => {  // use `{...}` curly braces since there is quite a lot more logic now than before.
  if (transitionDuration > 0) {
    enter.transition().duration(transitionDuration)
  }
  enter
    .attr('font-size', getFontSize)
    .attr(...)
    .text(getText)
}),
```

Do the same for the other areas.  Basically, we only run `.transition().duration(transitionDuration)` if it is larger than 0.

This leads to less code duplication, and also reduces adding an option that already has the features supported.

Let me know if it works!